pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -557,5 +561,41 @@ func (a *Bootstrap) Load(f asset.FileFetcher) (found bool, err error) {
 	}
 
 	a.File, a.Config = file, config
+	warnIfCertificatesExpired(a.Config)
 	return true, nil
 }
+
+// warnIfCertificatesExpired checks for expired certificates and warns if so
+func warnIfCertificatesExpired(config *igntypes.Config) {
+	expiredCerts := 0
+	for _, file := range config.Storage.Files {
+		if strings.HasSuffix(file.Path, ".crt") && file.Contents.Source != nil {
+			// ^data:text/plain;charset=utf-8;base64,LS0.+==
+			i := strings.Index(*file.Contents.Source, ",LS0")
+			if i == -1 {
+				continue
+			}
+			data, err := base64.StdEncoding.DecodeString((*file.Contents.Source)[i+1:])
+			if err != nil {
+				logrus.Debugf("Unable to decode certificate. %s", err.Error())
+				continue
+			}
+			block, _ := pem.Decode(data)
+			if block != nil {
+				cert, err := x509.ParseCertificate(block.Bytes)
+				if err == nil {
+					if time.Now().UTC().After(cert.NotAfter) {
+						logrus.Warnf("Bootstrap Ignition-Config Certificate %s expired at %s.", path.Base(file.Path), cert.NotAfter.Format(time.RFC3339))
+						expiredCerts++
+					}
+				} else {
+					logrus.Debugf("Unable to parse certificate. %s", err.Error())
```

Comment:
some information on which certificate would be helpful in the logs

Reply:
Fixed.